module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -897,12 +899,23 @@ public function getMyTransactions($patron)
             '/circulation/loans',
             $query
         ) as $trans) {
-            $date = date_create($trans->dueDate);
+            $date = new DateTime($trans->dueDate, new DateTimeZone('UTC'));
+            $loc = (new DateTime)->getTimezone();
+            $date->setTimezone($loc);
+
+            $dueStatus = false;
+            $tmpDueDate = $date->getTimestamp();
+
+            $now = time();
+            if ($now > $tmpDueDate) {
+                $dueStatus = 'overdue';
+            } elseif ($now > $tmpDueDate - (1 * 24 * 60 * 60)) {
+                $dueStatus = 'due';
+            }
             $transactions[] = [
-                'duedate' => date_format($date, "j M Y"),
-                'dueTime' => date_format($date, "g:i:s a"),
-                // TODO: Due Status
-                // 'dueStatus' => $trans['itemId'],
+                'duedate' => $date->format('d F Y'),
```

Comment:
Maybe we could here use `$this->dateConverter->convertToDisplayDate('U', $tmpDueDate)` and `$this->dateConverter->convertToDisplayTime('U', $tmpDueDate)` to take advantage of configurable date/time formats, similar to what we discussed on #2484. (And of course, please replace $tmpDueDate with the appropriate value if you choose to rename it per my suggestion above).